frontend/packages/noobaa-storage-plugin/src/components/noobaa-operator/resourceTable.tsx

```diff
@@ -0,0 +1,219 @@
+  ];
+};
+ResourceTableHeader.displayName = 'ResourceTableHeader';
+
+const getModelFromName = (name: string) => {
+  if (name === 'BucketClass') return NooBaaBucketClassModel;
+  return NooBaaBackingStoreModel;
+};
+
+const ResourceTableRow = (props) => {
+  const { obj, index, key, style } = props;
+  return (
+    <TableRow id={obj.metadata.uid} index={index} trKey={key} style={style}>
+      <TableData className={tableColumnClasses[0]}>
+        <ResourceLink
+          kind={referenceForModel(getModelFromName(obj.kind))}
+          name={obj.metadata.name}
+          namespace={obj.metadata.namespace}
+          title={obj.metadata.uid}
+        />
+      </TableData>
+      <TableData className={tableColumnClasses[1]}>
+        <ResourceLink kind={referenceForModel(getModelFromName(obj.kind))} name={obj.kind} />
+      </TableData>
+      <TableData className={tableColumnClasses[2]}>{obj.status.phase}</TableData>
+      <TableData className={classNames(tableColumnClasses[3], 'co-break-word')}>
+        <ResourceLink
+          kind="Namespace"
+          name={obj.metadata.namespace}
+          title={obj.metadata.namespace}
+        />
+      </TableData>
+      <TableData className={tableColumnClasses[4]}>{obj.metadata.creationTimestamp}</TableData>
+      <TableData className={tableColumnClasses[5]}>
+        <ResourceKebab
+          actions={Kebab.factory.common}
+          kind={referenceForModel(getModelFromName(obj.kind))}
+          resource={obj}
+        />
+      </TableData>
+    </TableRow>
+  );
+};
+
+const getFireHoseResources = (namespace: string) => {
+  const backingStore = {
+    kind: referenceForModel(NooBaaBackingStoreModel),
+    namespace,
+    namespaced: true,
+    prop: 'bs',
+    isList: true,
+  };
+  const bucketClass = {
+    kind: referenceForModel(NooBaaBucketClassModel),
+    namespaced: true,
+    namespace,
+    prop: 'bc',
+    isList: true,
+  };
+  return [backingStore, bucketClass];
+};
+
+const ListComponent = (props) => {
+  return (
+    <Table
+      {...props}
+      aria-label="NooBaa Resources"
+      Header={ResourceTableHeader}
+      Row={ResourceTableRow}
+      virtualize
+    />
+  );
+};
+
+const flattenResources = (resources) => {
+  const { bc, bs } = resources;
+  const bcData = !_.isEmpty(bc) ? bc.data : [];
+  const bsData = !_.isEmpty(bs) ? bs.data : [];
+  const data = bcData.concat(bsData);
+  const loaded = _.every(resources, (resource) => resource.loaded && _.isEmpty(resource.loadError));
+  if (!loaded) return [];
+  return data;
+};
+
+const getOperatorPathFromMatch = (matchObj: match) => {
+  const { url } = matchObj;
+  const parts = url.split('/');
+  return parts.slice(0, -2).join('/');
+};
+
+const MCGResourceList: React.FC<MCGResourceListProps> = (props) => {
+  const {
+    customData: { namespace },
+    match: matchObj,
+  } = props;
+
+  const operatorPath = getOperatorPathFromMatch(matchObj);
+
+  const createItems = {
+    'Backing Store': 'Backing Store',
```

Comment:
imo its better to use `NooBaaBackingStoreModel.label` and `NooBaaBucketClassModel.label` for every occurence